test/tests/interfacekernels/InterfaceSorption/tests

```diff
@@ -121,10 +116,243 @@
+    requirement = 'The system shall have the capability to enforce interfacial conditions based on the Sievert law in isothermal conditions during transient with unit scaling on both variables.'
+    type = 'Exodiff'
+    input = 'interface_sorption_transient.i'
+    cli_args = 'unit_scale=4
+                unit_scale_neighbor=6
+                Outputs/file_base=interface_sorption_transient_Sievert_non_ad_scaling_out'
+    exodiff = 'interface_sorption_transient_Sievert_non_ad_scaling_out.e'
+    mesh_mode = REPLICATED
+    recover = false
+  []
+  [Sievert_transient_ad_scaling]
+    requirement = 'The system shall have the capability to enforce interfacial conditions based on the Sievert law in isothermal conditions using automatic differentiation during transient with unit scaling on both variables.'
+    type = 'Exodiff'
+    input = 'interface_sorption_transient.i'
+    cli_args = 'unit_scale=4
+                unit_scale_neighbor=6
+                Kernels/u1_time_derivative/type=ADTimeDerivative
+                Kernels/u2_time_derivative/type=ADTimeDerivative
+                Kernels/temperature/type=ADTimeDerivative
+                Kernels/u1_diffusion/type=ADMatDiffusion
+                Kernels/u2_diffusion/type=ADMatDiffusion
+                Kernels/temperature/type=ADHeatConduction
+                InterfaceKernels/interface/type=ADInterfaceSorption
+                Materials/properties_1/type=ADGenericConstantMaterial
+                Materials/properties_1/type=ADGenericConstantMaterial
+                Materials/properties_2/type=ADGenericConstantMaterial
+                Postprocessors/flux_inner/type=ADSideDiffusiveFluxIntegral
+                Postprocessors/flux_outer/type=ADSideDiffusiveFluxIntegral
+                Outputs/file_base=interface_sorption_transient_Sievert_ad_scaling_out
+                Executioner/nl_abs_tol=5e-12'
+    exodiff = 'interface_sorption_transient_Sievert_ad_scaling_out.e'
+    mesh_mode = REPLICATED
+    recover = false
+  []
+  [Sievert_transient_non_ad_penalty_scaling]
+    requirement = 'The system shall have the capability to enforce interfacial conditions based on the Sievert law in isothermal conditions using a penalty-enforced flux balance during transient with unit scaling on both variables.'
+    type = 'Exodiff'
+    input = 'interface_sorption_transient.i'
+    cli_args = 'unit_scale=4
+                unit_scale_neighbor=6
+                InterfaceKernels/interface/use_flux_penalty=true
+                InterfaceKernels/interface/flux_penalty=1e1
+                Outputs/file_base=interface_sorption_transient_Sievert_non_ad_penalty_scaling_out'
+    exodiff = 'interface_sorption_transient_Sievert_non_ad_penalty_scaling_out.e'
+    # compared to Sievert_transient_non_ad_scaling because it is supposed to be equivalent,
+    # but the residuals are expected different with the penalty so the relative error tolerance is increased here
+    rel_err = 8.5e-2
```

Comment:
This is a relatively loose relative error for the comparison. Does tightening the solve tolerances in this test help achieve a more converged solution (and so allow the usage of a better diffing tolerance)?

This question is intended to cover all of the loose tolerances used in the later tests in this list as well.

Reply:
The `interfaceSorption` capability either uses the MatDiffusion kernel from the neighboring block to determine flux continuity, or it uses a penalty (as used in this test). When using the penalty approach, the residuals are different because the equation being solved is different. Tightening the solve tolerance do not help in this case for that reason. Changing the penalty value to the exact right amount *might* lead to the exact same result, but that would be a very case-dependent exercise and I don't think it's needed here.

The point of this test of to show that the penalty-based approach provides roughly the same values, even if it's different. 

Note that the different approaches are detailed in the documentation, and that I added a comment above each `rel_err = 8.5e-2` line in an attempt to clarify things. 

Another thing I can do is add another test for the penalty appraoch with it's own gold. This way, it directly tests the penalty-based approach with this new test AND makes sure it's not too far away from the other approach.

Reply:
Thank you for the context. This current test achieves both of those goals (usage of the penalty approach, and the fact that it achieves comparable results with the MatDiffusion approach). No standalone test is required just to check off penalty usage again.

Based on what you've said here, I can accept that this tolerance is necessary given the intended scope of the test.